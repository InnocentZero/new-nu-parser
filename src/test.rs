use crate::resolver::Resolver;
use crate::{compiler::Compiler, parser::Parser};
use std::path::Path;
use std::process::exit;

fn evaluate_example(fname: &Path) -> String {
    let mut compiler = Compiler::new();
    let contents = std::fs::read(fname).expect("We only run tests found by glob");

    let span_offset = compiler.span_offset();
    compiler.add_file(&fname.to_string_lossy(), &contents);

    let parser = Parser::new(compiler, span_offset);
    compiler = parser.parse();

    let mut result = compiler.display_state();

    if !compiler.errors.is_empty() {
        return result;
    }

    let mut resolver = Resolver::new(&compiler);
    resolver.resolve();
    result.push_str(&resolver.display_state());

    result
}

#[test]
fn test_node_output() {
    insta::glob!("../tests", "*.nu", |path| {
        insta::assert_snapshot!(evaluate_example(path));
    });
}
