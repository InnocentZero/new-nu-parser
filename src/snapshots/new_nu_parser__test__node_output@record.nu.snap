---
source: src/test.rs
expression: evaluate_example(path)
input_file: tests/record.nu
---
0: Name (1 to 2)
1: Int (4 to 5)
2: Garbage (8 to 9)
3: Garbage (10 to 11)
4: Garbage (11 to 12)
5: Garbage (13 to 13)
6: Garbage (13 to 13)
7: Garbage (13 to 13)
8: Record { pairs: [(NodeId(0), NodeId(1)), (NodeId(2), NodeId(4)), (NodeId(5), NodeId(7))] } (0 to 0)
9: Block(BlockId(0)) (0 to 13)

